internal/controller/streaming_manager.go

```diff
@@ -103,6 +119,8 @@ func (sm *streamManager) BootUpStreamAndReconnect(ctx context.Context) error {
 	}
 	allResourcesSnapshotted.Wait()
 	err = resourceLister.sendResourceSnapshotComplete()
+	dd.timeStarted = time.Now()
```

Comment:
You must do the same before the error return in case `sendClusterMetadata` returns an error, above.
Every `return` statement must be covered.
I'd have suggested to do this in a `defer`, but that wouldn't work since this method blocks on `<-Done()` below until the stream closes?!
What else can we do to ensure that these fields are updated before every `return`?

Reply:
I see what you are saying but have a few questions

- Lets say a goroutine fails part way through its requests and cancels context, do we really need to cover those cases since now this healthcheck will just restart the pod anyway? On top of that the it will just restart the whole process again in the backoff loop and reset thet timestamp in that deadlock object after initial bootup?